index.src.html

```diff
@@ -262,9 +279,10 @@
         with |promise|.
     1. Signal {{AudioDecoder/[[codec implementation]]}} to emit all <a>internal
         pending outputs</a>.
-    2. For each decoded audio data |output|, queue a task on the media element
-        task source to run the <a>Output AudioFrame</a> algorithm with |output|.
-    3. Queue a task on the media element task source to resolve |promise|.
+    2. For each decoded audio data |output|, queue a task on the
+        [=control thread=] event loop to run the <a>Output AudioFrame</a>
+        algorithm with |output|.
+    3. Queue a task on the [=control thread=] event loop to resolve |promise|.
```

Comment:
Multiple tasks, as we discussed. But we now have a for loop in the step 2, maybe we can consider resolving the promise in the task in 2, and doing the for loop in a single task? This would guarantee that running the actual implementation code for the (possibly multiple) "Output an AudioFrame" algorithm happens in one go, without anything possibly interrupting in between.

Of course as we discussed there is no 1:1 relationship between running a `decode` and getting an `AudioFrame`, it can be any cardinality, but maybe it's worth it considering doing this as a tight "for loop" in a single task instead of multiple tasks.

Reply:
Done. Applied only to AudioDecoder for now. LMK if you like it.

Reply:
I went ahead with the change to the other interfaces.